### PR TITLE
brainvision tests

### PR DESCRIPTION
**Pull request by massich**

### CODE REVIEW

**Comment by massich** [comment]
That's how I would have written it. But I miss understood the yield. (I think)

**Comment by massich** [comment]
cf69e87 should make the trick. And we get nice naming:

```sh
~/code/mne-python bvtests* 12s
(mne) ❯ pytest ./mne/io/brainvision/tests/test_brainvision.py -k foo -vs      
Coverage.py warning: --include is ignored because --source is set (include-ignored)
Test session starts (platform: linux, Python 3.6.8, pytest 4.4.0, pytest-sugar 0.9.2)
cachedir: .pytest_cache
rootdir: /home/sik/code/mne-python, inifile: setup.cfg
plugins: timeout-1.3.3, sugar-0.9.2, pudb-0.7.0, mock-1.10.3, faulthandler-1.5.0, cov-2.6.1
collecting ... ('Mk1=New Segment,,1,1,0,20131113161403794232\n', list([1384359243, 794231]), '2013-11-13 16:14:03 GM')

 mne/io/brainvision/tests/test_brainvision.py::test_foo[foo0] ✓                                                                                                                                                          25% ██▌       ('Mk1=STATUS,,1,1,0\n', None, 'unspecified')

 mne/io/brainvision/tests/test_brainvision.py::test_foo[foo1] ✓                                                                                                                                                          50% █████     ('Mk1=New Segment,,1,1,0,\n', None, 'unspecified')

 mne/io/brainvision/tests/test_brainvision.py::test_foo[foo2] ✓                                                                                                                                                          75% ███████▌  ('Mk1=New Segment,,1,1,0\n', None, 'unspecified')

 mne/io/brainvision/tests/test_brainvision.py::test_foo[foo3] ✓                                                                                                                                                         100% ██████████
----------------------------------------------------------------------------------- generated xml file: /home/sik/code/mne-python/junit-results.xml -----------------------------------------------------------------------------------


====================================================================================================== slowest 20 test durations ======================================================================================================
1.59s setup    mne/io/brainvision/tests/test_brainvision.py::test_foo[foo0]

(0.00 durations hidden.  Use -vv to show these durations.)

Results (10.80s):
       4 passed
      16 deselected

~/code/mne-python bvtests* 11s
(mne) ❯ 
```

File: mne/io/brainvision/tests/test_brainvision.py [hunk 1]
```diff
@@ -77,50 +77,81 @@ def test_orig_units(recwarn):
...
+}))
+# @pytest.mark.parametrize('newstring, expected', [
+#     ('Mk1=New Segment,,1,1,0,20131113161403794232\n', '2013-11-13 16:14:03 GMT'),  # noqa: E501
+#     ('Mk1=New Segment,,1,1,0,20070716122240937454\nMk2=New Segment,,2,1,0,20070716122240937455\n', '2007-07-16 12:22:40 GMT'),  # noqa: E501
+#     ('Mk1=STATUS,,1,1,0\n', 'unspecified'),
+#     ('Mk1=New Segment,,1,1,0,\n', 'unspecified'),
+#     ('Mk1=New Segment,,1,1,0\n', 'unspecified'),
+#     ('Mk1=New Segment,,1,1,0,00000000000304125000', 'unspecified'),
+#     ('Mk1=New Segment,,1,1,0,\nMk2=New Segment,,2,1,0,20070716122240937454\n', '2007-07-16 12:22:40 GMT'),  # noqa: E501
+# ])
```
Comment (massich) [inline mne/io/brainvision/tests/test_brainvision.py]: @sappelhoff can you finish to add the missing tests?

File: mne/io/brainvision/tests/test_brainvision.py [hunk 2]
```diff
@@ -77,50 +77,81 @@ def test_orig_units(recwarn):
...
+
+@pytest.fixture(scope='session')
+def _mocked_meas_date_data(tmpdir_factory):
+    """Helper funct. preparing the files for mocked_meas_date_file fixture."""
+
```
Comment (massich) [inline mne/io/brainvision/tests/test_brainvision.py]: maybe we don't need this empty line

File: mne/io/brainvision/tests/test_brainvision.py [hunk 3]
```diff
@@ -77,50 +77,81 @@ def test_orig_units(recwarn):
...
+@pytest.fixture(scope='session', params=[tt for tt in TEST_CASES])
+def mocked_meas_date_file(_mocked_meas_date_data, request):
+    """Return vmrk text and index of New Segment line for date tests."""
+    MEAS_DATE_LINE = 11
 
-    return tmp_vhdr_file, tmp_vmrk_file, lines, idx
+    vhdr_fname = _mocked_meas_date_data['vhdr_fname']
+    vmrk_fname = _mocked_meas_date_data['vmrk_fname']
+    eeg_fname = _mocked_meas_date_data['eeg_fname']
+    lines = _mocked_meas_date_data['lines']
```
Comment (massich) [inline mne/io/brainvision/tests/test_brainvision.py]: maybe there is a nicer way to unpack the dictionary. I started doing it with a for loop but it was looking even more ugly.

File: mne/io/brainvision/tests/test_brainvision.py [hunk 4]
```diff
@@ -77,50 +77,81 @@ def test_orig_units(recwarn):
...
-    pytest.param('Mk1=New Segment,,1,1,0,\nMk2=New Segment,,2,1,0,20070716122240937454\n', '2007-07-16 12:22:40 GMT'),  # noqa: E501
-])
-def test_dates(_get_date_test_data, newstring, expected):
-    """Test valid dates."""
-    tmp_vhdr_file, tmp_vmrk_file, lines, idx = _get_date_test_data
+    yield dict(vhdr_fname=vhdr_fname,
+               vmrk_fname=vmrk_fname,
+               eeg_fname=eeg_fname,
+               expected_meas_date=request.param['meas_date'],
+               expected_meas_date_repr=request.param['meas_date_repr'])
```
Comment (massich) [inline mne/io/brainvision/tests/test_brainvision.py]: we only need `vhrd_fname`. If so:
```py
    yield dict(fname=vhdr_fname,
               expected_meas_date=request.param['meas_date'],
               expected_meas_date_repr=request.param['meas_date_repr'])
```

File: mne/io/brainvision/tests/test_brainvision.py (hunk 3 above)
Comment (sappelhoff) [reply mne/io/brainvision/tests/test_brainvision.py]: We could use an `OrderedDict` and use 
```Python
vhdr_fname, vmrk_fname , eeg_fname, lines = _mocked_meas_date_data.values()
```

also: we don't need `eeg_fname`, so I'll cut that

File: mne/io/brainvision/tests/test_brainvision.py (hunk 4 above)
Comment (sappelhoff) [reply mne/io/brainvision/tests/test_brainvision.py]: yes, I'll change that to the reduced version :+1:

**Pull request merged**